extensions/src/org/pentaho/platform/web/http/api/resources/KarafResource.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright 2015 Pentaho Corporation. All rights reserved.
+ */
+
+package org.pentaho.platform.web.http.api.resources;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.APPLICATION_XML;
+
+import java.util.List;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+
+import org.codehaus.enunciate.jaxrs.ResponseCode;
+import org.codehaus.enunciate.jaxrs.StatusCodes;
+import org.pentaho.platform.osgi.KarafInstance;
+import org.pentaho.platform.osgi.KarafInstancePropertyDto;
+
+/**
+ * This service provides methods related to karaf instances
+ *
+ * @author tkafalas
+ */
+@Path( "/karaf" )
+public class KarafResource extends AbstractJaxRSResource {
```

Comment:
Yea want to see this reimagined as a generic service for the platform. Anyone should be able to add a service/port registration to be returned by this service. Karaf would be just one of the systems feeding ports into it. So probably need a ServiceRegistry class and a ServiceResource. Add the ability to add a Service, with name, ID, Description and used ports